Logging with build tags (#329)
gosnmp/gosnmp: README.md

  (see Usage below). `Connect` uses the `GoSNMP` struct;
  `gosnmp.Default` is provided for you to build on.
* GoSNMP no longer relies on **alouca/gologger** - you can use your
  logger if it conforms to the `gosnmp.LoggerInterface` interface; otherwise
  debugging will disabled.

```go
type LoggerInterface interface {
    Print(v ...interface{})
    Printf(format string, v ...interface{})
}
```
To enable logging, you must call gosnmp.NewLogger() function, and pass a pointer to your logging interface, for example with standard *log.Logger:

```go

> gosnmp.Default.Logger = NewLogger(log.New(os.Stdout, "", 0))

Reviewer: Shouldn't this be `gosnmp.NewLogger()`?

Author: Fixed

```
or
```go
g := &gosnmp.GoSNMP{
    ...

>     Logger:    NewLogger(log.New(os.Stdout, "", 0)),

Reviewer: Same here, `gosnmp.NewLogger()`?

Author: Fixed

}

```
You can completely remove the logging code from your application using the golang build tag "gosnmp_nodebug", for example:
```
go build -tags gosnmp_nodebug
```

> This will completely disable the logging of the gosnmp library, even if the logger interface is specified in the code. This will allow you to manage debug / release builds without changing the code, and will also improve performance. Whenever possible, always create release builds using this tag.

Reviewer: I don't agree with this recommendation. Logging is important for production use. Without good logs, diagnosing production problems is made more difficult.

```suggestion
This will completely disable the logging of the gosnmp library, even if the logger interface is specified in the code. This provides a small performance improvement.```

Author: I agree, I wanted to change that too. My original formulation looks ambiguous. Fixed.
